Tasks/AzureKeyVaultV2/operations/azure-arm-keyvault.ts

```diff
@@ -59,7 +59,8 @@ export class KeyVaultClient extends azureServiceClient.ServiceClient {
 
                 return response;
             } catch(error) {
-                if (retriableErrorCodes.indexOf(error.code) != -1 && ++retryCount < maxRetryCount) {
+                if ((retriableErrorCodes.indexOf(error.code) != -1 || error.toString().indexOf('Request timeout: ') != -1)
```

Comment:
Why is this not already workking? It has the ETIMEDOUT in it. Is the error.code something else?

Reply:
What is the value of `error.code` in cases when `error.toString().indexOf('Request timeout: ') != -1`?

Reply:
this.beginRequest(request) invokes
https://github.com/microsoft/azure-pipelines-tasks/blob/master/common-npm-packages/azure-arm-rest-v2/AzureServiceClientBase.ts#L105
that in turn invokes
https://github.com/microsoft/azure-pipelines-tasks/blob/master/common-npm-packages/azure-arm-rest-v2/webClient.ts#L47
and if error.code is not null there it will be printed to console:
https://github.com/microsoft/azure-pipelines-tasks/blob/master/common-npm-packages/azure-arm-rest-v2/webClient.ts#L77
But since there were no such messages in the pipeline logs then it means that error.code was empty, so we have to check error message to catch this particular case.

Reply:
I am on the same page as Martin there. However, I know we are probably not able to influence the shape of `error` object here. Is it possible to reproduce it locally and check the content of `error`? We might be able to find something more reliable than comparing error messages.

Reply:
Found the sources of the place where this error is being thrown - https://github.com/microsoft/typed-rest-client/blob/master/lib/HttpClient.ts#L386

Reply:
So there is no other info there passed to Error c-tor beside it's message:
new Error('Request timeout: ' + info.options.path).
So we can't use anything else beside message to catch this particular case.